feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package optics_power_and_bias_current_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/telemetry"
+	"github.com/openconfig/ygot/ygot"
+)
+
+const (
+	transceiverType        = telemetry.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_TRANSCEIVER
+	sleepDuration          = time.Minute
+	minOpticsPower         = -30.0
+	maxOpticsPower         = 10.0
+	minOpticsHighThreshold = 1.0
+	maxOpticsLowThreshold  = -1.0
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Topology:
+//   ate:port1 <--> port1:dut:port2 <--> ate:port2
+//
+//  Sample CLI command to get telemetry using gmic:
+//   - gnmic -a ipaddr:10162 -u username -p password --skip-verify get \
+//      --path /components/component --format flat
+//   - gnmic tool info:
+//     - https://github.com/karimra/gnmic/blob/main/README.md
+//
+
+func TestOpticsPowerBiasCurrent(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	transceivers := findComponentsByType(t, dut, transceiverType)
+	t.Logf("Found transceiver list: %v", transceivers)
+	if len(transceivers) == 0 {
+		t.Fatalf("Get transceiver list for %q: got 0, want > 0", dut.Model())
+	}
+
+	for _, transceiver := range transceivers {
+		t.Logf("Validate transceiver: %s", transceiver)
+		component := dut.Telemetry().Component(transceiver)
+
+		if !component.MfgName().Lookup(t).IsPresent() {
```

Comment:
Prefer to reduce the levels of nesting by eschewing else.

```
for _, transceiver := range transceivers {
  ...
  if !component.MgfName().Lookup(t).IsPresent() {
    t.Logf(...)
    continue
  }
  mfgName := ...
  ...
}
```

Reply:
done

Reply:
done